app/assets/javascripts/repository_columns/index.js

```diff
@@ -0,0 +1,247 @@
+/* global I18n HelperModule animateSpinner */
+/* no-restricted-globals */
+var RepositoryColumns = (function() {
+  function initColumnTypeSelector() {
+    $('[name="repository_column[data_type]"]')
+      .on('click', function() {
+        $('.column-type').hide();
+        $('[data-column-type="' + $(this).val() + '"]').show();
+      });
+  }
+
+  // function replaceListItem(column) {
+  //   $('.list-group-item[data-id="' + column.id + '"]')
+  //     .find('span.pull-left').text(column.name);
+  // }
+  //
+  // function initTagInput() {
+  //   $('[data-role="tagsinput"]').tagsinput({
+  //     maxChars: GLOBAL_CONSTANTS.NAME_MAX_LENGTH,
+  //     trimValue: true
+  //   });
+  // }
+  //
+  // function processResponse(form, action, modalID) {
+  //   form.on('ajax:success', function(e, data) {
+  //     switch (action) {
+  //       case 'destroy':
+  //         removeElementFromDom(data);
+  //         break;
+  //       case 'create':
+  //         break;
+  //       case 'update':
+  //         replaceListItem(data);
+  //         break;
+  //       default:
+  //         location.reload();
+  //     }
+  //     HelperModule.flashAlertMsg(data.message, 'success');
+  //     animateSpinner(null, false);
+  //     if (modalID) {
+  //       $(modalID).modal('hide');
+  //     }
+  //   }).on('ajax:error', function(e, xhr) {
+  //     animateSpinner(null, false);
+  //     if (modalID) {
+  //       if (xhr.responseJSON.message.hasOwnProperty('repository_list_items')) {
+  //         var message = xhr.responseJSON.message['repository_list_items'];
+  //         $('.dnd-error').remove();
+  //         $('#manageRepositoryColumn ').find('.bootstrap-tagsinput').after(
+  //           "<i class='dnd-error'>" + message + "</i>"
+  //         );
+  //       } else {
+  //         var field = { "name": xhr.responseJSON.message };
+  //         $(form).renderFormErrors('repository_column', field, true, e);
+  //       }
+  //     } else {
+  //       HelperModule.flashAlertMsg(xhr.responseJSON.message, 'danger');
+  //     }
+  //   });
+  //   if (modalID) {
+  //     form.submit();
+  //   }
+  // }
+
+  // @TODO refactor that
+  function initEditCoumnModal() {
+    var modalID = '#manageRepositoryColumn';
+    // var colRadID = '#repository_column_data_type_repositorylistvalue';
+    // var tagsInputID = '[data-role="tagsinput"]';
+    // var formID = '[data-role="manage-repository-column-form"]';
+
+    $('[data-action="edit"]').off('click').on('click', function() {
+      var editUrl = $(this).closest('li').attr('data-edit-url');
+      $.get(editUrl, function(data) {
+        $(data.html).appendTo('body').promise().done(function() {
```

Comment:
Let's clean this promise. This code already sync, remove all promises

Reply:
:+1: